### ISSUE TITLE
[ENH] Past predictions over train set

### ISSUE THREAD

**Issue by IlyasMoutawwakil** [alan-turing-institute/sktime#1380]
> **Is your feature request related to a problem? Please describe.**

I was wondering why can't we get past predictions (by that I mean predictions made by the model on the train set) in sktime, actually one of the most fundamental ways to compare models is to compare their performance on train set relatively to test set performance and see if they're over-fitting: if the model performance on train set is way better than it is one the test/validation set. But in sktime we can only (I guess) see future predictions because of the fh argument that starts at 1 (as the first prediction after train set) which means we can only compare y_pred to y_test.

> **Describe the solution you'd like**

I think this needs a refactoring of how forecasters treat the `ForecastHorizon`. either forecasters should be able to accept negative values in their `fh` argument (kinda off) or maybe (as in statsmodels) the new forecasts should start at `len(y_train)`.

> **Additional context**

This makes a big difference for models that can be overfitted aka models with lots of hyper-parameters such as prophet where adding more and more seasonalities with high order can make your future predictions worse but your past predictions better. Actually all models that allow multiple seasonalities (BATS and TBAT) can over-fit easily but with this you'll be able to see it with your eyes.

**Comment by IlyasMoutawwakil** [comment]
Please tell me if their's a way around it.

**Comment by mloning** [comment]
Hi @IlyasMoutawwakil, most forecasters should support in-sample predictions, you can specify them with the forecasting horizon, e.g. `ForecastingHorizon(y_train.index, is_relative=False)`. 

Check out the section "3.2.2 The Detrender as pipeline component" in the notebook here: https://github.com/alan-turing-institute/sktime/blob/main/examples/01_forecasting.ipynb - let me know if this helps!

**Comment by IlyasMoutawwakil** [comment]
Thanks a lot that's what I've been looking for.

**Comment by aiwalter** [comment]
@IlyasMoutawwakil backtesting works with `evaluate`, see also notebooks.
https://www.sktime.org/en/latest/api_reference/auto_generated/sktime.forecasting.model_evaluation.evaluate.html

**Comment by fkiraly** [comment]
@ilyasmoutawwakil, not sure whether in-sample forecasts are what you are looking fore - those shouldn't be used for backtesting!

**Comment by IlyasMoutawwakil** [comment]
Hmm I'm quite lost here, what I'm trying to do is as I explained above. In other words: the predictions over training data from the forecaster that have seen the train data (all of it) which is not exactly what an expanding window does.

**Comment by aiwalter** [comment]
Yes then just do insample prediction

**Comment by fkiraly** [comment]
yes, agreed with @aiwalter here. 

And yes, `Detrender` does this - it's not totally obvious, so we should probably add this to the forecasting tutorial notebook - perhaps a section 1.6, frequently asked "how to" questions? Would you be up to adding that, @IlyasMoutawwakil?

**Comment by aiwalter** [comment]
I think `Detrender` is just needed for residuals. I think for residuals would be nice to have a getter in the base class which does it as a shorthand with `forecaster.resid_` and the getter just uses the `Detrender`

**Comment by fkiraly** [comment]
there already is a discussion about it in #1105 - @aiwalter, @IlyasMoutawwakil, perhaps you can add your opinions how, if at all, in-sample prediction and/or residuals should look like?

The current suggestion is:
* `fit_predict` for in-sample predictions, shorthand
* `Detrender` for residuals, but also a tutorial section

`predict_residuals` or similar as you just suggested is an alternative?
